### PR TITLE
feat(tasks-new): Edit task message and due date in TaskForm

### PR DESCRIPTION
**Pull request by shahzadaziz**
- Ability to edit and update Task `message` and `dueDate`. 
- Error handling if edit fails

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @shahzadaziz has signed the CLA. Thanks for the pull request!

File: src/elements/content-sidebar/ActivitySidebar.js [hunk 1]
```diff
@@ -585,6 +592,7 @@ class ActivitySidebar extends React.PureComponent<Props, State> {
             createTask,
             getApproverWithQuery,
             getAvatarUrl,
+            id: '',
```
Comment (shahzadaziz) [inline src/elements/content-sidebar/ActivitySidebar.js]: This is a required form field prop.

File: src/elements/content-sidebar/activity-feed/task-form/TaskForm.js [hunk 2]
```diff
@@ -126,31 +131,44 @@ class TaskForm extends React.Component<Props, State> {
...
         const { message, approvers, dueDate, isValid } = this.state;
         const dueDateString = dueDate && dueDate.toISOString();
 
         if (!isValid) return;
 
         this.setState({ isLoading: true });
-        createTask(message, approvers, taskType, dueDateString, this.handleCreateSuccess, this.handleCreateError);
+
+        if (editMode === TASK_EDIT_MODE_EDIT && editTask) {
+            editTask(
```
Comment (booc0mtaco) [inline src/elements/content-sidebar/activity-feed/task-form/TaskForm.js]: I think `editTask` and `createTask` should have the same signature. Do we want to update `createTask` to use an object? or should `editTask` be flattened?

Comment (shahzadaziz) [reply src/elements/content-sidebar/activity-feed/task-form/TaskForm.js]: Object might be easier to deal with a flowType.

File: src/api/Feed.js [hunk 3]
```diff
@@ -533,6 +533,7 @@ class Feed extends Base {
                 }
             },
             errorCallback: (e: ElementsXhrError, code: string) => {
+                this.updateFeedItem({ isPending: false }, task.id);
```
Comment (jeremypress) [inline src/api/Feed.js]: Are we showing any visual indication that the update failed?

Comment (shahzadaziz) [reply src/api/Feed.js]: Yes. TaskModal has an error prop which renders an inlineError.

**Pull request merged**